parseutil/parseutil.go

```diff
@@ -401,3 +402,54 @@ func ParseAddrs(addrs interface{}) ([]*sockaddr.SockAddrMarshaler, error) {
 
 	return out, nil
 }
+
+func SafeParseIntRange(in interface{}, min int64, max int64) (int64, error) {
```

Comment:
This is not a problem of your making, but it can be really, really confusing at this point to figure out which function to use from among many similar ones (for instance, would I use the direct variant, safe variant, or neither?). Can you please add comments to these functions explaining what they do? I realize it is relatively obvious but it would be good to declare intentions so at a glance people can figure out if it's the right one.

Reply:
Generally the direct should only be used if the value for sure is a sliced value. If you're unsure (especially in an API where either a direct value or a list of values could be provided), I'd think its probably best to use `ParseIntSlice`, hence why that was given the shorter name over the `*Direct*` variant (and is inclusive of the latter's functionality).

There's no corresponding safe version of the `*Direct*` variant as I'm not sure I personally see much use in it, as it was mostly a helper towards the more permissive `ParseInt` version that someone else might find useful (given its a shared library).

I've added doc texts to all methods, clarifying that when the result of a non-Safe variant is called, the caller would have to manually validate range constraints to cast the result to a more restrictive numeric type, but that the Safe variants handle this for the caller.